lib/ansible/plugins/filter/core.py

```diff
@@ -100,7 +100,10 @@ def to_bool(a):
 
 
 def to_datetime(string, format="%Y-%m-%d %H:%M:%S"):
-    return datetime.datetime.strptime(string, format)
+    try:
+        return datetime.datetime.strptime(string, format)
+    except ValueError:
+        return False
```

Comment:
this changes current default behaviour, I would reject as is, but allow a toggle to change the behaviour explicitly:

```
to_datetime(string, format, on_error=fail|ignore|etc)
```

Reply:
I will add it like this